lib/pipeline.js

```diff
@@ -499,6 +487,46 @@ class PipelineModel extends BaseModel {
         return admin;
     }
 
+    /**
+     * This function deletes admins who does not have proper
+     * GitHub permission, and returns an proper admin.
+     * @method getFirstAdmin
+     * @return {Promise}
+     */
+    async getFirstAdmin() {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
+            try {
+                // Lazy load factory dependency to prevent circular dependency issues
+                // https://nodejs.org/api/modules.html#modules_cycles
+                /* eslint-disable global-require */
+                const UserFactory = require('./userFactory');
+                /* eslint-enable global-require */
+                const factory = UserFactory.getInstance();
+
+                // eslint-disable-next-line no-await-in-loop
+                const user = await factory.get({
+                    username,
+                    scmContext: this.scmContext
+                });
+
+                // eslint-disable-next-line no-await-in-loop
+                const permission = await user.getPermissions(this.scmUri);
+
+                if (permission.push) {
+                    return user;
+                }
+
+                throw username;
+            } catch (usernameNotAdmin) {
```

Comment:
why do you need `try - catch` here ? Instead of throwing in above line just do `delete this.admins[usernameNotAdmin]` there.

Reply:
`try - catch` is used to catch application or system error, so it might be not needed for this situation. I removed `try - catch`. Thank you!